server/src/serializers/TripSerializer.js

```diff
@@ -1,13 +1,48 @@
+import CommentSerializer from "./CommentSerializer.js"
+
 class TripSerializer {
   static getSummary(trip) {
-    const allowedAttributes = ["id", "continent", "country", "city", "title","numberOfDays", "description"]
+    const allowedAttributes = [
+      "id",
+      "continent",
+      "country",
+      "city",
+      "title",
+      "numberOfDays",
+      "description",
+    ]
+
+    let serializedTrip = {}
+    for (const attribute of allowedAttributes) {
+      serializedTrip[attribute] = trip[attribute]
+    }
+    return serializedTrip
+  }
+
+  static async getDetails(trip) {
+    const allowedAttributes = [
+      "id",
+      "continent",
+      "country",
+      "city",
+      "title",
+      "numberOfDays",
+      "description",
+    ]
 
     let serializedTrip = {}
     for (const attribute of allowedAttributes) {
       serializedTrip[attribute] = trip[attribute]
     }
+    const comments = await trip.$relatedQuery("comments")
+    let serializedComments = []
+    for (let i = 0; i < comments.length; i++) {
+      const comment = await CommentSerializer.getDetails(comments[i])
+      serializedComments.push(comment)
+    }
+    serializedTrip.comments = serializedComments
```

Comment:
We can clean this up a bit by making use of a `.map` iteration here for serializing the comments. Not completely necessary to update since this looks like it works just fine, but it would make this look a lot nicer. It's a _slightly_ different from the other patterns we've seen, but this is also a _slightly_ more advanced use for nested serializers! We can do something like:
```
serializedTrip.comments = await Promise.all(
  comments.map((comment) => {
    return CommentSerializer.getDetails(comment)
  }
)
```